int_tests/test_chat.py

```diff
@@ -0,0 +1,90 @@
+import time
+import unittest
+
+from vectara.core import RequestOptions
+from vectara.factory import Factory
+from vectara import CoreDocument, CoreDocumentPart, SearchCorporaParameters, KeyedSearchCorpus, \
+    ContextConfiguration, CustomerSpecificReranker, GenerationParameters, CitationParameters, \
+    ChatParameters
+
+
+class TestChat(unittest.TestCase):
+
+    def setUp(self):
+        self.addCleanup(self.cleanup)
+        self.client = Factory().build()
+        self.client.corpora.create(name="test-chat", key="test-chat")
+        time.sleep(30)
+        test_search_1_document = CoreDocument(
+            id="my-doc-id",
+            document_parts=[
+                CoreDocumentPart(
+                    text="Robot Utility Models are trained on a diverse set of environments and objects, and then can "
+                         "be deployed in novel environments with novel objects without any further data or training.",
+                )
+            ],
+        )
+        self.client.documents.create("test-chat", request=test_search_1_document)
+
+        self.search_params = SearchCorporaParameters(
+            corpora=[
+                KeyedSearchCorpus(
+                    corpus_key="test-chat",
+                    metadata_filter="",
+                    lexical_interpolation=1,
+                )
+            ],
+            context_configuration=ContextConfiguration(
+                sentences_before=2,
+                sentences_after=2,
+            ),
+            reranker=CustomerSpecificReranker(
+                reranker_id="rnk_272725719"
+            ),
+        )
+        self.generation_params = GenerationParameters(
+            response_language="eng",
+            citations=CitationParameters(
+                style="none",
+            ),
+            enable_factual_consistency_score=True,
+        )
+        self.chat_params = ChatParameters(store=True)
+        self.request_options = RequestOptions(timeout_in_seconds=100)
+
+    def test_chat(self):
```

Comment:
Can we test for some of the edge cases?
1. different search parameters (including none provided)
2. different generation parameters (including none provided)